l1-contracts/src/governance/GSE.sol

```diff
@@ -128,88 +220,106 @@ contract GSECore is IGSECore, Ownable {
    */
-  function deposit(address _attester, address _withdrawer, bool _onCanonical)
+  function deposit(address _attester, address _withdrawer, bool _onBonus)
     external
     override(IGSECore)
     onlyRollup
   {
-    address instanceAddress = msg.sender;
-    bool isCanonical = getCanonical() == instanceAddress;
-    require(!_onCanonical || isCanonical, Errors.GSE__NotCanonical(instanceAddress));
+    bool isMsgSenderLatestRollup = getLatestRollup() == msg.sender;
+
+    // If _onBonus is true, then msg.sender must be the latest rollup.
+    require(!_onBonus || isMsgSenderLatestRollup, Errors.GSE__NotLatestRollup(msg.sender));
 
-    // Ensure that we are not already attesting on the specific
+    // Ensure that we are not already attesting on the rollup
     require(
-      !isRegistered(instanceAddress, _attester),
-      Errors.GSE__AlreadyRegistered(instanceAddress, _attester)
+      !isRegistered(msg.sender, _attester), Errors.GSE__AlreadyRegistered(msg.sender, _attester)
     );
-    // Ensure that if we are canonical, we are not already attesting on the canonical
+
+    // Ensure that if we are the latest rollup, we are not already attesting on the bonus instance.
     require(
-      !isCanonical || !isRegistered(CANONICAL_MAGIC_ADDRESS, _attester),
-      Errors.GSE__AlreadyRegistered(CANONICAL_MAGIC_ADDRESS, _attester)
+      !isMsgSenderLatestRollup || !isRegistered(BONUS_INSTANCE_ADDRESS, _attester),
+      Errors.GSE__AlreadyRegistered(BONUS_INSTANCE_ADDRESS, _attester)
     );
 
-    if (_onCanonical) {
-      instanceAddress = CANONICAL_MAGIC_ADDRESS;
-    }
+    // Set the recipient instance address, i.e. the one that will receive the attester.
+    // From above, we know that if we are here, and _onBonus is true,
+    // then msg.sender is the latest instance,
+    // but the user is targeting the bonus address.
+    // Otherwise, we use the msg.sender, which we know is a registered rollup
+    // thanks to the modifier.
+    address recipientInstance = _onBonus ? BONUS_INSTANCE_ADDRESS : msg.sender;
 
+    // Add the attester to the instance's checkpointed set of attesters.
     require(
-      instances[instanceAddress].attesters.add(_attester),
-      Errors.GSE__AlreadyRegistered(instanceAddress, _attester)
+      instances[recipientInstance].attesters.add(_attester),
+      Errors.GSE__AlreadyRegistered(recipientInstance, _attester)
     );
 
-    instances[instanceAddress].configOf[_attester] = AttesterConfig({withdrawer: _withdrawer});
+    instances[recipientInstance].configOf[_attester] = AttesterConfig({withdrawer: _withdrawer});
 
-    if (delegation.getDelegatee(instanceAddress, _attester) == address(0)) {
-      delegation.delegate(instanceAddress, _attester, instanceAddress);
-    }
-
-    delegation.increaseBalance(instanceAddress, _attester, DEPOSIT_AMOUNT);
+    delegation.delegate(recipientInstance, _attester, recipientInstance);
```

Comment:
Why have you changed the logic here 👀 

Edit: Looked, and ye the only update that can be made over here is from `withdraw` or `delegate` and those only exists as cases when there is a deposit, so when it is not possible to run this if already in, then it is also the case that the delegatee is `address(0)` when you run it.